### PR TITLE
Card #332 Fix daytime ranges deleting, remove slice

### PR DESCRIPTION
**Pull request by Erbidium**

### CODE REVIEW

**Review by EugeneValko — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 1]
```diff
@@ -241,14 +241,26 @@ public async Task<AvailabilitySlotDto> GetAvailabilitySlotById(long id)
...
+                        .Select(dto => dto.Id);
+                var deletedExclusionDates = await _context.ExclusionDates.Where(date =>
+                        date.ScheduleId == scheduleId &&
+                        updatedExclusionDateIds.All(updatedDateId => updatedDateId != date.Id))
+                    .ToListAsync();
+                var deletedDayTimeRanges = await _context.DayTimeRanges
+                    .Where(range =>
+                        updatedExclusionDateIds.Any(updatedExclusionDateId =>
+                            updatedExclusionDateId == range.ExclusionDateId) && 
+                            updatedDayTimeRangeIds.All(updatedDayTimeRangeId => updatedDayTimeRangeId != range.Id))
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: you can move this as a predicate to the delete method
Without data loading

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 1 above)
Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: or add cascade deleting

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 1 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Cascade deleting is already added, but when merging daytime ranges, some ranges can be deleted(when a larger range from both sides is added to a smaller one) and this is for cases like this. I will move this as predicate

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Fixed

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**